### ISSUE TITLE
The decision to drop drivers/PPDs is a poorly planned and thought out one.

### ISSUE THREAD

**Issue by eightballocto** [apple/cups#6191]
As the title says, I believe the decision to drop support for drivers and PPDs is a very poorly planned out and one that wasn't thought out very well. I doubt my input will have any bearing on the end result of the project's trajectory, but I just wanted to drop an issue here to express my thoughts, as well as ask for some potential alternatives or solutions for when this change comes to pass. 
_For reference, I am running Arch Linux and am using a Canon LBP6030w printer. The last driver made available for CUPS was uploaded to the AUR in 2020. It works, but if this change goes through as it is now, I'm in trouble. Suggestions on how to deal with this are appreciated._

I've briefly scrubbed through the issue thread on when this decision was made back in 2018 and I already saw that it was controversial to begin with, and I can see why. There haven't been any real proof-of-concepts provided for the proposed alternative for "legacy" printers, nor does OP back up their claims of "security issues" or "PPDs holding back the user experience" with any kind of evidence, it's sort of just a "trust me bro" kind of thing. One person made a very good point in that it seems like the maintainers of this project are trying to wash their hands of the issue of keeping up with PPD support and tossing the responsibility to someone else, when in fact, this should have been part of the job description of CUPS all along.

In the response, one of the maintainers tries to reassure this person that this is a "future" plan, and that they're "working with developers" to support printers that would otherwise be incompatible with CUPS without the drivers. However, this doesn't address the elephant in the room, which is that many people are using printers that are working on drivers that are essentially one-off releases, and are likely never to be updated. I personally don't have the time, nor energy to reverse-engineer this printer in order to make it work with a system that should have already supported it in the first place, but here we are.

As it stands now, there's no clear plans as to how this "migration" will work, nor have there been any updates (as far as I can see) on the progression of alternative systems. It's been 4 years since this announcement, and I'd think that would be more than enough time to _at the very least_ write a rudimentary memo or brief announcement.

I beg of you (maintainers), to either reverse-course on this change, or to come up with a clearer compromise, or even just make more details known so that we know what exactly we're dealing with besides a "depreciated" warning.

**TL;DR if this decision is made without a clear roadmap for "legacy" printers, both known and lesser-known, then you're hanging out *a lot* of people to dry, and this could have a massive ripple effect, especially for those who can't afford to *not* update.**

**Comment by dkosovic** [comment]
Arch Linux doesn't use Apple CUPS anymore, it uses [OpenPrinting CUPS](https://github.com/OpenPrinting/cups), so you posted this issue in the wrong place.

When CUPS 3.0 is released, classic CUPS drivers can be handled by [pappl-retrofit](https://github.com/OpenPrinting/pappl-retrofit) or other PAPPL based Printer Apps for legacy printers.

The upcoming CUPS 2.5 will still handle PPD files.

**Comment by eightballocto** [comment]
> Arch Linux doesn't use Apple CUPS anymore, it uses [OpenPrinting CUPS](https://github.com/OpenPrinting/cups), so you posted this issue in the wrong place.
> 
> When CUPS 3.0 is released, classic CUPS drivers can be handled by [pappl-retrofit](https://github.com/OpenPrinting/pappl-retrofit) or other PAPPL based Printer Apps for legacy printers.
> 
> The upcoming CUPS 2.5 will still handle PPD files.

I realized this shortly before you replied, I guess speak of the devil. Thanks for pointing me in the right direction, apologies for not doing my homework. Closing this issue.
